src/main/java/randoop/main/Minimize.java

```diff
@@ -0,0 +1,1183 @@
+   * @param verboseOutput whether to produce verbose output
+   * @return true if minimization produced a (possibly unchanged) file that fails the same way as
+   *     the original file
+   */
+  public static boolean mainMinimize(
+      String filePath, String classPath, int timeoutLimit, boolean verboseOutput) {
+    System.out.println("Reading and parsing: " + filePath);
+
+    // Read and parse input Java file.
+    CompilationUnit compUnit;
+    try (FileInputStream inputStream = new FileInputStream(filePath)) {
+      compUnit = JavaParser.parse(inputStream);
+    } catch (ParseException e) {
+      System.err.println("Error parsing Java file: " + filePath);
+      System.err.println(e);
+      return false;
+    } catch (IOException e) {
+      System.err.println("Error reading Java file: " + filePath);
+      System.err.println(e);
+      return false;
+    }
+
+    System.out.println("Getting expected output.");
+
+    // Find the package name of the input file if it has one.
+    String packageName = null;
+    try {
+      PackageDeclaration classPackage = compUnit.getPackage();
+      if (classPackage != null) {
+        packageName = classPackage.getPackageName();
+      }
+    } catch (NoSuchElementException e) {
+      // No package declaration.
+    }
+
+    File originalFile = new File(filePath);
+    // Create a new string to represent the new file name.
+    String minimizedFileName =
+        new StringBuilder(filePath).insert(filePath.lastIndexOf('.'), "Minimized").toString();
+    File minimizedFile = new File(minimizedFileName);
+
+    // Rename the overall class to [old class name][suffix].
+    String orgClassName = FilenameUtils.removeExtension(originalFile.getName());
+    new ClassRenamer().visit(compUnit, new String[] {orgClassName, "Minimized"});
+
+    // Write the compilation unit to the minimized file.
+    writeToFile(compUnit, minimizedFile);
+
+    // Compile the Java file and check that the exit value is 0.
+    if (compileJavaFile(minimizedFile, classPath, packageName, timeoutLimit) != 0) {
+      System.err.println("Error when compiling file " + filePath + ". Aborting.");
+      return false;
+    }
+
+    String runResult = runJavaFile(minimizedFile, classPath, packageName, timeoutLimit);
+
+    // expectedOutput is a map from method name to failure stack trace
+    // with line numbers removed.
+    Map<String, String> expectedOutput = normalizeJUnitOutput(runResult);
+
+    System.out.println("Minimizing: " + filePath);
+
+    // Minimize the Java test suite.
+    minimizeTestSuite(
+        compUnit,
+        packageName,
+        minimizedFile,
+        classPath,
+        expectedOutput,
+        timeoutLimit,
+        verboseOutput);
+
+    // Cleanup: simplify variable type names and sort the import statements.
+    compUnit =
+        simplifyVariableTypeNames(
+            compUnit, packageName, minimizedFile, classPath, expectedOutput, timeoutLimit);
+    sortImports(compUnit);
```

Comment:
`simplifyVariableTypeNames` should call `sortImports`.  It's natural for `simplifyVariableTypeNames` to do this work, and this way the client does not have to remember to do so.